lib/web/apiserver.go

```diff
@@ -898,14 +903,22 @@ func (h *Handler) githubLoginWeb(w http.ResponseWriter, r *http.Request, p httpr
 }
 
 func (h *Handler) githubLoginConsole(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
-	h.log.WithField("auth", "github").Debug("Console login start.")
+	logger := h.log.WithField("auth", "github")
+	logger.Debug("Console login start.")
+
+	generalErr := trace.AccessDenied(ssoLoginConsoleErr)
```

Comment:
Even though this seems to reduce the error handling (if that's the intent), it will obscure the actual point of error emission and might complicate analysis as several places would be likely candidates as the error source - it's better to emit the error at each point with the corresponding `trace` error constructor.

Reply:
Revised so that each error returns its own trace err so tracing is closer to origin of actual error